sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -134,7 +134,8 @@ case class AdaptiveSparkPlanExec(
     CoalesceShufflePartitions(context.session),
     // `OptimizeShuffleWithLocalRead` needs to make use of 'AQEShuffleReadExec.partitionSpecs'
     // added by `CoalesceShufflePartitions`, and must be executed after it.
-    OptimizeShuffleWithLocalRead
+    OptimizeShuffleWithLocalRead,
+    PropagateEmptyPartitions
```

Comment:
I had try to put this `PropagateEmptyPartitions` before other rules, but it will cause 20 TESTS FAILED in  `AdaptiveQueryExecSuite`, since other rules do not support the new `EmptyPartitionSpec` for now. So I conservatively put it at the end.